File: problems/0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums, target):
        numandindex={}
        for i in range(len(nums)):
            other=target-nums[i]
            if other in numandindex:
                return[numandindex[other],i]
            numandindex[nums[i]]=i
solution=Solution()
ans=solution.twoSum([2,7,11,15],9)
print(ans)

File: problems/0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> seen;
        int complement;
        for (int i = 0; i < int(nums.size()); i++){
            complement=target-nums[i];
            if (seen.count(complement)){return vector<int> {seen[complement],i};}
            seen[nums[i]]=i;
        }  
        return {}; 
    }
};
